### PR TITLE
Feature/41 document the required component configuration structure

### PR DESCRIPTION
**Pull request by AdrianaAC**

### CODE REVIEW

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 1]
```diff
@@ -1,24 +1,290 @@
...
+    <tr>
+      <td>arrayValues</td>
+      <td>
+        <span style={{ color: 'green' }}>
+          {' '}
+          array {'<'}string{'>'}
+        </span>
+      </td>
+      <td>Set of properties used in the component</td>
+      <td>string / number - ❌ singleSelect / multiSelect - ✅</td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:213]: Make it like:

      <td>❌ if string or number<br/>✅ if singleSelect or multiSelect</td>

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 2]
```diff
@@ -1,24 +1,290 @@
...
+      <td>undefined</td>
+      <td>['small', 'medium', 'large'], ['outlined', 'rounded'], ['column', 'row']</td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:215]: This is confusing, it should be an array of values, but it shows three arrays of values. Change this to a single example array.

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 3]
```diff
@@ -1,24 +1,290 @@
...
+          <span style={{color:"green"}}>boolean</span>
+      </td>
+      <td>
+        Component is deletable, true if yes, false if no.
+      </td>
+      <td>
+        ✅
+      </td>
+      <td>
+       	undefined
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:153]: Check the code to understand what is the default value of this and update accordingly.

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 4]
```diff
@@ -1,24 +1,290 @@
...
+          <span style={{color:"green"}}>boolean</span>
+      </td>
+      <td>
+        Component is editable, true if yes, false if no.
+      </td>
+      <td>
+        ✅
+      </td>
+      <td>
+        	undefined
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:135]: Check the code to understand what is the default value of this and update accordingly.

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 5]
```diff
@@ -1,24 +1,290 @@
...
+          <span style={{color:"green"}}>boolean</span>
+      </td>
+      <td>
+        Children components allowed, true if yes, false if no.
+      </td>
+      <td>
+        ✅
+      </td>
+      <td>
+      undefined
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:117]: Check the code to understand what is the default value of this and update accordingly.

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 6]
```diff
@@ -1,24 +1,290 @@
...
+      <td>
+        Identifies the type of component. Types cannot be shared between components.
+      </td>
+      <td>
+        ✅
+      </td>
+      <td>
+        undefined
+      </td>
+      <td></td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:59]: Missing example value

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 7]
```diff
@@ -1,24 +1,290 @@
...
+      <td>
+        An array that identifies each property needed by the component.
+      </td>
+      <td>
+        ❌
+      </td>
+      <td>
+        {"[]"}
+      </td>
+      <td></td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:77]: Missing example value

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 8]
```diff
@@ -1,24 +1,290 @@
...
+      <td>
+        ```json
+          {
+          "children": undefined,
+          "editable": undefined,
+          "delete": undefined,
+        }
+        ```
+      </td>
+      <td></td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:101]: Missing example value

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 9]
```diff
@@ -1,24 +1,290 @@
...
+      <td>
+        Children components allowed, true if yes, false if no.
+      </td>
+      <td>
+        ✅
+      </td>
+      <td>
+      undefined
+      </td>
+      <td></td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:119]: Missing example value

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 10]
```diff
@@ -1,24 +1,290 @@
...
+      <td>
+        Component is editable, true if yes, false if no.
+      </td>
+      <td>
+        ✅
+      </td>
+      <td>
+        	undefined
+      </td>
+      <td></td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:137]: Missing example value

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 11]
```diff
@@ -1,24 +1,290 @@
...
+      <td>
+        Component is deletable, true if yes, false if no.
+      </td>
+      <td>
+        ✅
+      </td>
+      <td>
+       	undefined 
+      </td>
+      <td></td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:155]: Missing example value

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 12]
```diff
@@ -1,24 +1,290 @@
...
+
+<table>
+  <thead>
+    <tr>
+      <td>**Name**</td>
+      <td>**Type**</td>
+      <td>**Description**</td>
+      <td>**Required**</td>
+      <td>**Default**</td>
+      <td>**Possible**</td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:170]: Change to "example"

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 13]
```diff
@@ -1,24 +1,290 @@
...
+  <tbody>
+    <tr>
+      <td>type</td>
+      <td>
+        <span style={{ color: 'green' }}>string</span>
+      </td>
+      <td>Type of the property</td>
+      <td>✅</td>
+      <td>undefined</td>
+      <td>string / number / singleSelect / multiSelect</td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:182]: Move this to the type and add an example.
Use typescript annotation, type could be: "string" | "number" | "singleSelect" | "multiSelect"

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 14]
```diff
@@ -1,24 +1,290 @@
...
+    </tr>
+    <tr>
+      <td>name</td>
+      <td>
+        <span style={{ color: 'green' }}>string</span>
+      </td>
+      <td>The name of property inside the component</td>
+      <td>✅</td>
+      <td>undefined</td>
+      <td></td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:192]: Missing example value

File: packages/lumina-docs/src/stories/docs/Components.mdx [hunk 15]
```diff
@@ -1,24 +1,290 @@
...
+    </tr>
+    <tr>
+      <td>label</td>
+      <td>
+        <span style={{ color: 'green' }}>string</span>
+      </td>
+      <td>The label showed on the Lumina editor UI</td>
+      <td>✅</td>
+      <td>undefined</td>
+      <td></td>
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/docs/Components.mdx:202]: Missing example value

**Review by JWebCoder — CHANGES_REQUESTED** [review]
Please check the review

File: packages/lumina-docs/src/stories/docs/Components.mdx (hunk 13 above)
Comment (AdrianaAC) [reply packages/lumina-docs/src/stories/docs/Components.mdx:182]: This is the type of the props, not the component, so it should be kept in this type

**Review by JWebCoder — APPROVED** [review]

**Pull request merged**